test/awaitility.go

```diff
@@ -0,0 +1,25 @@
+package test
+
+import (
+	"time"
+)
+
+type Awaitility struct {
+	timeout time.Duration
+	before  time.Time
+}
+
+func WaitWithTimeout(timeout time.Duration) Awaitility {
+	return Awaitility{timeout: timeout, before: time.Now()}
+}
+
+func (a Awaitility) Until(do func() error) error {
+	err := do()
+	if err == nil {
+		return nil
+	} else if a.before.Before(time.Now().Add(-a.timeout)) {
+		return err
+	}
+	time.Sleep(500 * time.Millisecond)
+	return a.Until(do)
+}
```

Comment:
go way of doing this - https://play.golang.org/p/TJS0qxJElrj
less logic and more magic ;-)

Reply:
yes, this was my initial thought as well. Thanks @nurali-techie for the simplified example!

Reply:
now `before` is not used anywhere.  I guess .. no need of struct and can be done with func only.

Reply:
Yes, this is what happens when I'm doing several things on multiple branches at once :-( 
removed in 61fe716 
Thanks @nurali-techie